### DISCUSSION TITLE
When posting a request to the server, what is the format for specifying logit biases?

### DISCUSSION THREAD

**Discussion by phasiclabs** [ggml-org/llama.cpp#4392] [Q&A] [answered]
Does anyone know what the format for specifying logit biases? I can't find any info on it!

I've been posting json like this to the lllama.cpp server, and everything works apart from the 'logit-bias'
{
	'prompt': "<prompt goes here>", 
	'n_predict': 128, 
	'temperature': 0.9, 
	'mirostat': 2, 
	'mirostat-ent': 5, 
	'image_data': [{'data': '<image data here>', 'id': 12}], 
	'stream': True, 
	'logit-bias': [[<logit1>, 100], [<logit2>, 100], [<logit3>, 100]]
}

From looking at the server code, it seems to expect an array of arrays, but when I breakpoint it it find the logit-bias array but can't handle the values inside, so does nothing.

**Comment by KerfuffleV2** [comment, marked as answer]
I checked the code. It looks like it want an array of token id/bias pairs _or_ instead of bias you can use a bool and false is the same as setting negative infinity as the bias. So that would be `[[TOKEN_ID, BIAS], [TOKEN_ID, BIAS]]` or more specifically `[[123,1.1], [345,-1.0]]`.

**Reply by phasiclabs** [reply]
Thanks - that is actually the format I was trying to make work - my original post looked like that, but had the token id stripped out by Github for some reason, because I put them inside less than/greater than symbols.

I'll try debugging it again, but it seems to find the outer array, but then then detects inner arrays as 'objects' rather than arrays, so this test fails :

if (el.is_array() && el.size() == 2 && el[0].is_number_integer())

**Reply by phasiclabs** [reply]
Turns out it works fine in you use 'logit_bias' (as documented), and not 'logit-bias' as I had been using!
